frontend/packages/ceph-storage-plugin/src/components/modals/delete-snapshot/delete-snapshot.tsx

```diff
@@ -0,0 +1,62 @@
+import * as React from 'react';
+
+import { HandlePromiseProps, withHandlePromise } from '@console/internal/components/utils';
+import {
+  ModalBody,
+  ModalComponentProps,
+  ModalSubmitFooter,
+  ModalTitle,
+  createModalLauncher,
+} from '@console/internal/components/factory';
+
+import { Form } from '@patternfly/react-core';
+import { VolumeSnapshotModel } from '../../../models';
+import { k8sKill } from '@console/internal/module/k8s';
+
+export const DeleteSnapshotModal = withHandlePromise((props: DeleteSnapshotModalProps) => {
+  const { close, cancel, resource, errorMessage, inProgress } = props;
+
+  const submit = (event: React.FormEvent<EventTarget>) => {
+    event.preventDefault();
+    props
+      .handlePromise(k8sKill(VolumeSnapshotModel, resource))
```

Comment:
destructure it with the rest in line 17.

Reply:
ack